src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -3286,25 +3285,107 @@ void MavlinkReceiver::update_rx_stats(const mavlink_message_t &message)
 		PX4_WARN("Max remote components of %u used up", MAX_REMOTE_COMPONENTS);
 		_warned_component_states_full_once = true;
 	}
+
+
+	if (_message_statistics_enabled) {
```

Comment:
What about putting all of that new code in a separate function?

Then you can make some early returns, e.g.:

```
if (!_received_msg_stats) {
    return;
}
```

And that way you can keep the levels of indentation a bit more sane.

Reply:
Works for me, updated.